src/main/java/com/singh/astha/notification/service/dtos/common/NotificationRequest.java

```diff
@@ -13,12 +16,17 @@
 @AllArgsConstructor
 public class NotificationRequest {
 
+    @NotNull
+    @Min(1)
     private Long userId;
 
+    @NotBlank
     private String templateId;
 
+    @NotNull
     private HashMap<String, String> titlePlaceholder;
 
+    @NotNull
     private HashMap<String, String> bodyPlaceHolders;
```

Comment:
Ideally we should allow these to be null but if they are not null then they should be not empty.

We can let this go for now. For future create a task to create custom validation annotation `@NullOrNotEmpty`